contracts/contracts/strategies/CurvePoolBooster.sol

```diff
@@ -0,0 +1,175 @@
+        uint256 bridgeFee,
+        uint256 additionalGasLimit
+    ) external onlyOperator {
+        require(campaignId != 0, "Campaign not created");
+
+        // Cache current rewardToken balance
+        uint256 extraTotalRewardAmount = IERC20(rewardToken).balanceOf(
+            address(this)
+        );
+
+        // Approve the total reward amount to the campaign manager
+        require(extraTotalRewardAmount > 0, "No reward to manage");
+
+        // Approve the total reward amount to the campaign manager
+        IERC20(rewardToken).approve(
+            campaignRemoteManager,
+            extraTotalRewardAmount
+        );
+
+        // Manage the campaign
+        ICampaingRemoteManager(campaignRemoteManager).manageCampaign{
+            value: bridgeFee
+        }(
+            ICampaingRemoteManager.CampaignManagementParams({
+                campaignId: campaignId,
+                rewardToken: rewardToken,
+                numberOfPeriods: 0,
+                totalRewardAmount: extraTotalRewardAmount,
+                maxRewardPerVote: 0
+            }),
+            targetChainId,
+            additionalGasLimit
+        );
+    }
+
+    function manageNumberOfPeriods(
+        uint8 extraNumberOfPeriods,
+        uint256 bridgeFee,
+        uint256 additionalGasLimit
+    ) external onlyOperator {
+        require(campaignId != 0, "Campaign not created");
+
+        ICampaingRemoteManager(campaignRemoteManager).manageCampaign{
+            value: bridgeFee
+        }(
+            ICampaingRemoteManager.CampaignManagementParams({
+                campaignId: campaignId,
+                rewardToken: rewardToken,
+                numberOfPeriods: extraNumberOfPeriods,
+                totalRewardAmount: 0,
+                maxRewardPerVote: 0
+            }),
+            targetChainId,
+            additionalGasLimit
+        );
+    }
+
+    function manageRewardPerVote(
+        uint256 newMaxRewardPerVote,
+        uint256 bridgeFee,
+        uint256 additionalGasLimit
+    ) external onlyOperator {
+        require(campaignId != 0, "Campaign not created");
+
+        ICampaingRemoteManager(campaignRemoteManager).manageCampaign{
+            value: bridgeFee
+        }(
+            ICampaingRemoteManager.CampaignManagementParams({
+                campaignId: campaignId,
+                rewardToken: rewardToken,
+                numberOfPeriods: 0,
+                totalRewardAmount: 0,
+                maxRewardPerVote: newMaxRewardPerVote
+            }),
+            targetChainId,
+            additionalGasLimit
+        );
+    }
+
+    function setCampaignId(uint256 _campaignId) external onlyOperator {
+        campaignId = _campaignId;
+    }
+
+    function setOperator(address _newOperator) external onlyGovernor {
+        operator = _newOperator;
+    }
+
+    function sendETH(address receiver) external onlyOperator {
+        payable(receiver).transfer(address(this).balance);
```

Comment:
> to recover any accidentally sent tokens?
Not really. This contract will hold a bit of ETH, as we need to pay a bit when calling the CampaignRemoteManager (CCIP fees). So it is in the situation where the pool booster will not be used anymore (end of services) and the operator what to claim the remaining ETH. 

And using a payable function is not the best (imo) because a surplus of ETH is sent back to this contract, not to the sender.

What do you think of keeping `onlyOperator` modifier then?

Reply:
IMO it is fine, as there will only be small amount of ETH on the contract whose purpose is to fuel the gas for transactions and that is not part of the user funds.

Reply:
Yes, exactly, only to fuel gas transactions, nothing more.